lib/rules/prefer-const.js

```diff
@@ -357,6 +357,9 @@ module.exports = {
         const shouldMatchAnyDestructuredVariable = options.destructuring !== "all";
         const ignoreReadBeforeAssign = options.ignoreReadBeforeAssign === true;
         const variables = [];
+        let reportCount = 0;
+        const firstIdentifier = sourceCode.ast.tokens.find(node => node.type === "Identifier");
```

Comment:
I'm confused about this line. It looks like this line will grab the first identifier token in the program, without consideration for scope.

If you add a few tests that have some nested scopes, you might see some issues. For example:

```js
const foo = 1;   // First identifier token is on this line

function someFunc() {
    let a = 0,
        b = 1;

    bar(a, b);
    // Both a and b should be made const in this case, but are they detected properly?
}
```

Reply:
@platinumazure thanks for the feedback, good point, didnt think about this but yes this is detected properly.  The line you commented on is just an initializer and then compares current with previous name.  I will add tests for the various scoping options as well.